src/bidiMapper/domains/events/EventManager.ts

```diff
@@ -79,6 +79,24 @@ export class EventManager implements IEventManager {
     events: string[],
     contextIds: BrowsingContext.BrowsingContext[] | null
   ): Promise<void> {
-    throw new Error('EventManager.unsubscribe is not implemented');
+    for (let event of events) {
+      if (contextIds === null) this._unsubscribe(event, null);
+      else {
+        for (let contextId of contextIds) {
+          this._unsubscribe(event, contextId);
+        }
+      }
+    }
+  }
+
+  private _unsubscribe(
+    event: string,
+    contextId: BrowsingContext.BrowsingContext | null
+  ): void {
+    if (!this._subscriptions.has(contextId)) return;
+
+    this._subscriptions.get(contextId)!.delete(event);
```

Comment:
it's better to use `?` instead of !. `!` is unsafe typecasting whereas would be checked at runtime.